### ISSUE TITLE
[Question] Is there a way to download the playwright firefox browser without running `npm install`?

### ISSUE THREAD

**Issue by zMrKrabz** [microsoft/playwright#5543]
I packaged my app with `pkg`, but when users run the file, they get `Failed to launch firefox because executable does not exist`. Is there a way I can catch this error, then download the browser?

**Comment by zMrKrabz** [comment]
I found [this](https://github.com/microsoft/playwright/blob/master/src/install/installer.ts) script, but am unsure how to import it into my script

**Comment by zMrKrabz** [comment]
Found the `install-from-github` script [here](https://github.com/microsoft/playwright/blob/791c8dabe1130f3f5156471caf1c57de3898bd9a/install-from-github.js#L30), but when I copy it, I get `TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined`.
Code:
```js
import { installBrowsersWithProgressBar } from 'playwright/lib/install/installer';

(async () => {
	await installBrowsersWithProgressBar();
})();
```

**Comment by aslushnikov** [comment]
> Is there a way I can catch this error, then download the browser?

@zMrKrabz Nah, there's no official way to do this, but you can definitely hack it in!

> Found the install-from-github script here

This is the right way to hack it :) not sure why you have the error though - should be working just fine.

**Comment by zMrKrabz** [comment]
My current hack is running a batch script that does `npm install`, My second comment is the code I ran and got error with. When I looked at the compiled javascript, the function requires 2 arguments while the typescript from the repository does not.

**Comment by aslushnikov** [comment]
> My current hack is running a batch script that does npm install

@zMrKrabz Good enough!

> When I looked at the compiled javascript, the function requires 2 arguments while the typescript from the repository does not.

This code has been recently changed, so it's not what you have in Playwright 1.8.0. When investigating the codebase, use tags that mark releases. For example, to see the code for v1.8.1: https://github.com/microsoft/playwright/tree/v1.8.1

**Comment by zMrKrabz** [comment]
Fixed issues, am currently on `1.9.0`. I think my main issue was that as I was traversing the code on Github, it switched commits for me? Weird. Here is my code to fix.

```ts
import { installBrowsersWithProgressBar } from 'playwright/lib/install/installer';
(async () => {
    await installBrowsersWithProgressBar();
})();
```

If you're writing tests, here is is how tested that the library works:

```ts
import { installBrowsersWithProgressBar } from 'playwright/lib/install/installer';
import { registryDirectory } from 'playwright/lib/utils/registry';

describe('Playwright download', () => {
	it('Installs browser in cache folder', async () => {
		await fs.rm(registryDirectory, { recursive: true });
		await installBrowsersWithProgressBar();
		await firefox.launch();
	});
});
```ts
